JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesClient.java

```diff
@@ -150,21 +187,49 @@ public void bulkInitTimeSeries(List<List<String>> dataIRIs, List<List<Class<?>>>
 	 * @param timeUnit
 	 * @param srid
 	 * @param conn
+	 * @param type
+	 * @param durations
+	 * @param units
 	 */
-    public void bulkInitTimeSeries(List<List<String>> dataIRIs, List<List<Class<?>>> dataClass, List<String> timeUnit, Integer srid, Connection conn) {
-        // create random time series IRI
-    	List<String> tsIRIs = new ArrayList<>(dataIRIs.size());
-    	
-    	for (int i = 0; i < dataIRIs.size(); i++) {
-    		String tsIRI = TimeSeriesSparql.ns_kb + "Timeseries_" + UUID.randomUUID();
-    		tsIRIs.add(i, tsIRI);
-    	}
-    	
+    public void bulkInitTimeSeries(List<List<String>> dataIRIs, List<List<Class<?>>> dataClass, List<String> timeUnit, Integer srid, Connection conn, List<Type> type, List<Duration> durations, List<ChronoUnit> units) {
```

Comment:
should this be private?

Reply:
this is the public facing method used for bulk instantiation. why would we want it to be private?

Reply:
This contains srid and the duration parameters (which will be an unlikely combination to be called outside), I thought this is the private method we agreed on that will be called by the other public facing ones?